ansible/roles/postgres/tasks/main.yml

```diff
@@ -0,0 +1,234 @@
+  delegate_to: "{{ item }}"
+  hostname:
+    name: "postgres_master"
+  with_items:
+    - "{{ postgres_master }}"
+
+- name: '[Slave1] Set hostname'
+  run_once: true
+  become: true
+  delegate_to: "{{ item }}"
+  hostname:
+    name: "postgres_slave_1"
+  with_items:
+    - "{{ postgres_slave_1 }}"
+
+- name: '[Slave2] Set hostname'
+  run_once: true
+  become: true
+  delegate_to: "{{ item }}"
+  hostname:
+    name: "postgres_slave_2"
+  with_items:
+    - "{{ postgres_slave_2 }}"
+
+- name: '[All] Add nodes to /etc/hosts'
+  become: true
+  blockinfile:
+    path: /etc/hosts
+    block: |
+      {{ hostvars[postgres_master].ec2_private_ip_address }} postgres_master
+      {{ hostvars[postgres_slave_1].ec2_private_ip_address }} postgres_slave_1
+      {{ hostvars[postgres_slave_2].ec2_private_ip_address }} postgres_slave_2
+
+- name: '[Master] Perform the initdb command'
+  run_once: true
+  delegate_to: "{{ item }}"
+  command: "/usr/pgsql-12/bin/postgresql-12-setup initdb"
+  environment:
+    PGSETUP_INITDB_OPTIONS: '--encoding=UTF-8'
+  args:
+    creates: /var/lib/pgsql/12/data/PG_VERSION
+  become: true
+  with_items:
+    - "{{ postgres_master }}"
+
+- name: '[Master] Make the PostgreSQL server can be connected with TCP/IP'
+  run_once: true
+  delegate_to: "{{ item }}"
+  lineinfile:
+    path: /var/lib/pgsql/12/data/postgresql.conf
+    regexp: "^listen_addresses = .*$"
+    line: "listen_addresses = '*'"
+  become: true
+  with_items:
+    - "{{ postgres_master }}"
+
+- name: '[Master] Configure network access'
+  run_once: true
+  delegate_to: "{{ item }}"
+  blockinfile:
+    path: /var/lib/pgsql/12/data/pg_hba.conf
+    content: |
+      # authorized receiver/sender connection
+      host all         all 0.0.0.0/0          md5
+      # trusted slaves connections
+      host replication all postgres_slave_1 trust
+      host replication all postgres_slave_2 trust
+  become: true
+  with_items:
+    - "{{ postgres_master }}"
+
+- name: '[Master] Start the PostgreSQL 12 service and make it running on rebooting'
+  run_once: true
+  delegate_to: "{{ item }}"
+  systemd:
+    name: postgresql-12
+    enabled: yes
+    state: started
+  become: true
+  with_items:
+    - "{{ postgres_master }}"
+
+- name: '[Master] Create a PostgerSQL user named mq'
+  run_once: true
+  become: yes
+  delegate_to: "{{ item }}"
+  become_user: postgres
+  postgresql_user:
+    login_unix_socket:
+    login_user: postgres
+    name: mq
+    password: pass
+  with_items:
+    - "{{ postgres_master }}"
+
+- name: '[Master] Create the mq database'
+  run_once: true
+  become: true
+  delegate_to: "{{ item }}"
+  become_user: postgres
+  postgresql_db:
+    login_unix_socket:
+    login_user: postgres
+    name: mq
+    owner: mq
+    state: present
+  with_items:
+    - "{{ postgres_master }}"
+
+- name: '[Master] Create jobs table'
+  run_once: true
+  become: true
+  delegate_to: "{{ item }}"
+  become_user: postgres
+  postgresql_table:
```

Comment:
I think I would create the table as part of the application, not the service setup, but that's a minor thing